### ISSUE TITLE
Flags Field in Transaction 

### ISSUE THREAD

**Issue by sera-korbit** [XRPLF/xrpl4j#300]
Hi, XRP Team!

I have a question about the flags field of a transaction.
According to the docs, the flags field is optional and there is no need to set FLAG as it defaults to 0.
However, if there is no flag in the payment transaction, an error has occured.
Error is  "tecPATH_PARTIAL".
I've inserted all possible flag options such as NO_DIRECT_RIPPLE, LIMIT_QUALITY, UNSET. The error occurred except for the tfParitalPayment flag.
Is there a proper way to construct PAYMENT Type Transaction without setting FLAG?
My code is below.

        Payment payment = null;
        payment = Payment.builder()
                .account(fromAddress)
                .amount(IssuedCurrencyAmount
                        .builder()
                        .issuer(Address.of(issuer))
                        .currency(currencyCode)
                        .value(amount)
                        .build()
                )
                .destination(Address.of(toAddress))
                .sequence(sequence)
                .fee(openLedgerFee)
                .flags(Flags.PaymentFlags.builder().tfPartialPayment(true).build())
                .signingPublicKey(rippleWallet.publicKey())
                .lastLedgerSequence(lastLedgerSequence)
                .build();

**Comment by sappenin** [comment]
Hi @sera-korbit - thanks for your message. I'm not sure flags are your problem here. You're getting a `tecPATH_PARTIAL` error from rippled because the entire payment couldn't be made -- specifically from the docs, this means `The transaction failed because the provided [paths](https://xrpl.org/paths.html) did not have enough liquidity to send the full amount.`

You can google around for some scenarios that would cause that (e.g., try this [query](https://www.google.com/search?q=tecPATH_PARTIAL&oq=tecPATH_PARTIAL)). 

However, can you share more about the trustline setup? E.g., what are the Trustline configurations of (1) the issuer (2) the sender and (3) the receiver?

**Comment by sera-korbit** [comment]
Hi! @sappenin

**Comment by sera-korbit** [comment]
Hi @sappenin
I was trying to send SOLO(sologenic). Issuer and currencyCode are below.
Issuer : rsoLo2S1kiGeCcn6hCUXVrCpGMWLrRrLZz, currencyCode : 534F4C4F00000000000000000000000000000000

I've searched also that error message and what it means. At first I thought there might be a problem with the balance in fromAddress. The balance may not be sufficient.

However, even though fromAddress had enough balance (both XRP and SOLO) to send SOLO, I got that message except for putting the "tfPartialPayment" flag.
